EosGround/radioDriver.py

```diff
@@ -0,0 +1,103 @@
+import time
+
+from digi.xbee.devices import XBeeDevice
+from digi.xbee.devices import RemoteXBeeDevice
+from digi.xbee.devices import XBee64BitAddress
+
+from EosLib.packet.packet import Packet
+import EosLib.packet.definitions
+import EosLib.packet.packet
+import EosLib.packet.transmit_header
+
+import psycopg2
+from config.config import config
+import datetime
+
+global sequence_number
+sequence_number = 0
+
+conn_params = config('database.ini')  # gets config params
+conn = psycopg2.connect(**conn_params)  # gets connection object
+conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)  # sets up auto commit
+cursor = conn.cursor()  # creates cursor
+
+# sets up digi
+PORT = "COM5"
+device = XBeeDevice(PORT, 9600)
+device.open()
+
+
+# function called when data is received
+def data_receive_callback(xbee_message):
+    packet = Packet.decode(xbee_message.data)
+
+    packet_type = packet.data_header.data_type
+    packet_sender = packet.data_header.sender
+    packet_priority = packet.data_header.priority
+    packet_generate_time = packet.data_header.generate_time
+    packet_sequence_number = packet.transmit_header.send_seq_num
+    packet_timestamp = packet.transmit_header.send_time
+    packet_body = packet.body.decode()
+    time_arrived = datetime.datetime.now()
+
+    print(packet_body)
+
+    cursor.execute(
+        """
+        INSERT INTO receive_table (packet_type, packet_sender, packet_priority, packet_generate_time, packet_sequence_number, packet_timestamp, packet_body, time_arrived) VALUES 
+        (%s,%s,%s,%s,%s,%s,%s,%s)
+        """, (
+        packet_type, packet_sender, packet_priority, packet_generate_time, packet_sequence_number, packet_timestamp,
+        packet_body, time_arrived)
+    )
+    conn.commit()
+
+
+#  function called anytime new data is put into database
+def send_command():
+    global sequence_number
+    cursor.execute("""
+    SELECT * FROM "transmit_table"
+    ORDER BY "id" DESC
+    LIMIT 1;
+    """)
+
+    row = cursor.fetchall()[0] # gets last row in table
```

Comment:
instead of doing `LIMIT 1` and `fetchall`, remove the limit clause and do `for row in cursor:`
this is more efficient because the server only has to execute 1 query and cache 1 result set instead of having to do it N times

Reply:
If "for row in cursor:" is implemented then it will get all the rows. We only need the last row hence the LIMIT 1.